### PR TITLE
Added methods for next tab, previous tab, getting the tab-index, and activating by tab index.

### PR DESCRIPTION
**Pull request by dsandber**
I finally got around to doing the regression tests you requested in #13

### CODE REVIEW

**Comment by freekmurze** [comment]
Conflicts. Could you rebase this againt the current master?

**Comment by dsandber** [comment]
Can I do it through "resolve conflicts" or is that problematic for some reason?

**Comment by freekmurze** [comment]
Do it with what suits you best.

**Comment by dsandber** [comment]
Resolved conflicts.

**Comment by vesper8** [comment]
this is beautiful! can't wait to use this in combination with keyboard arrows. thanks for adding it!

File: src/components/Tab.vue [hunk 1]
```diff
@@ -29,9 +29,8 @@
             },
 
             hash() {
-                return this.id ?
-                    '#' + this.id :
-                    '#' + this.name.toLowerCase().replace(/ /g, '-');
+                let hash_suffix = this.id ? this.id : this.name.toLowerCase().replace(/ /g, '-');
```
Comment (freekmurze) [inline src/components/Tab.vue:32]: The previous version is more readable to me. Could you revert this?

File: src/components/Tabs.vue [hunk 2]
```diff
@@ -97,10 +101,43 @@
                 this.$emit('changed', { tab: selectedTab });
 
                 this.activeTabHash = selectedTab.hash;
+                if (setLocationHash) {
```
Comment (freekmurze) [inline src/components/Tabs.vue:104]: Let's let the code breath a little. Add a newline before this line.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Nice PR!

File: src/components/Tab.vue (hunk 1 above)
Comment (sebastiandedeyne) [inline src/components/Tab.vue:32]: Wouldn't `/\s/g` be a better choice here?

File: src/components/Tabs.vue [hunk 3]
```diff
@@ -97,10 +101,43 @@
...
+
+                this.selectTab(href, null, setLocationHash);
+            },
+
+            getActiveTabIndex() {
+                const tab = this.findTab(this.activeTabHash);
+                return this.tabs.indexOf(tab);
+            },
+
+            nextTab(wrap = false) {
```
Comment (sebastiandedeyne) [inline src/components/Tabs.vue:123]: Could `wrap` be renamed to `cycle` everywhere?

File: src/components/Tabs.vue [hunk 4]
```diff
@@ -97,10 +101,43 @@
...
             },
 
+            selectTabByIndex(index, setLocationHash) {
+                const tab = this.tabs[index];
+                const href = tab.hash;
+
+                this.selectTab(href, null, setLocationHash);
+            },
+
+            getActiveTabIndex() {
```
Comment (sebastiandedeyne) [inline src/components/Tabs.vue:118]: I'd make a computed property out of this

File: src/components/Tabs.vue [hunk 5]
```diff
@@ -97,10 +101,43 @@
...
+                const tab = this.findTab(this.activeTabHash);
```
Comment (sebastiandedeyne) [inline src/components/Tabs.vue:119]: And move this to it's own `activeTab` computed prop

File: src/components/Tabs.vue [hunk 6]
```diff
@@ -97,10 +101,43 @@
...
+                return this.tabs.indexOf(tab);
+            },
+
+            nextTab(wrap = false) {
+                let index = this.getActiveTabIndex();
```
Comment (sebastiandedeyne) [inline src/components/Tabs.vue:124]: `index` doesn't need to be mutated here, could be a `const`.

```js
const index = this.getActiveTabIndex() + 1;
```

File: src/components/Tabs.vue [hunk 7]
```diff
@@ -97,10 +101,43 @@
...
+                index++;
+                if (index === this.tabs.length) {
+                    index = wrap ? 0 : index - 1;
+                }
+                this.selectTabByIndex(index, true);
+            },
+
+            previousTab(wrap = false) {
+                let index = this.getActiveTabIndex();
```
Comment (sebastiandedeyne) [inline src/components/Tabs.vue:133]: Same as `nextTab`

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]
Looking good! Requested a few more changes before we merge :)

**Comment by sebastiandedeyne** [comment]
Gonna close this for now, feel free to spin up a new PR with the requested feedback if you still want this merged in, no priority for us :)

**Pull request closed without merging**